lib/trino-filesystem-s3/src/test/java/io/trino/filesystem/s3/TestS3FileSystemAwsS3.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.s3;
+
+import io.airlift.units.DataSize;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import static java.util.Objects.requireNonNull;
+
+@EnabledIfEnvironmentVariable(named = "AWS_ACCESS_KEY", matches = ".+")
+@EnabledIfEnvironmentVariable(named = "AWS_SECRET_KEY", matches = ".+")
+@EnabledIfEnvironmentVariable(named = "S3_REGION", matches = ".+")
+@EnabledIfEnvironmentVariable(named = "S3_BUCKET", matches = ".+")
```

Comment:
any typo here (eg AWS_ACCESS_KEY instead of AWS_ACCESS_KEY_ID) silently disabled the test.

use pom profiles for that

with ci.yml + maven profiles it's much easier to document what SHOULD be run and then test either succeeds or fails to configure

to make it even more robust, we use `||`s, not `&&`s to disable tests and that's also why CI_SKIP_SECRETS_PRESENCE_CHECKS was added to mean "run all the tests defined"